Studio-Lovelies/GG-JointJustice-Unity
# Introduce separate `PlayMode Playthrough` test step and flag

## Summary
This PR splits the automated playthrough tests into a separate job so they can run in parallel to `EditMode`, `PlayMode Scripts` and `PlayMode Scenes` tests.

## Before/after screenshots and/or animated gif
Before: 
<img width="299" alt="image" src="https://user-images.githubusercontent.com/1689033/192135778-9e3b32b3-76b3-4bf2-9bd7-d68ac4075914.png">

After: 
<img width="287" alt="image" src="https://user-images.githubusercontent.com/1689033/192136453-68fda78b-1d39-4da9-9229-5db6835ed3a9.png">

Additionally, test suits were labeled incorrectly (`PlayMode Scripts` were reported as `PlayMode Scenes` and vice-versa), which has also been addressed.

## Testing instructions
1. Execute the build pipeline for this branch (or a sub-branch thereof)
2. Notice automated playthrough tests running in parallel to other test suits
3. Check that CodeCov reports the new `PlayMode Playthrough` flag

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - #246 
  - #247
